src/Stratis.Bitcoin.Features.Api/ApiSettings.cs

```diff
@@ -44,29 +41,32 @@ public class ApiSettings
         /// Initializes an instance of the object from the node configuration.
         /// </summary>
         /// <param name="nodeSettings">The node configuration.</param>
-        public ApiSettings(NodeSettings nodeSettings)
+        /// <param name="defaults">The default values.</param>
+        public ApiSettings(NodeSettings nodeSettings, ApiSettingsDefaults defaults = null)
         {
             Guard.NotNull(nodeSettings, nameof(nodeSettings));
 
             this.logger = nodeSettings.LoggerFactory.CreateLogger(typeof(ApiSettings).FullName);
 
             TextFileConfiguration config = nodeSettings.ConfigReader;
 
-            this.UseHttps = config.GetOrDefault("usehttps", false);
-            this.HttpsCertificateFilePath = config.GetOrDefault("certificatefilepath", (string)null);
+            defaults = defaults ?? new ApiSettingsDefaults(nodeSettings.Network);
+
+            this.UseHttps = config.GetOrDefault("usehttps", defaults.UseHttps);
+            this.HttpsCertificateFilePath = config.GetOrDefault("certificatefilepath", defaults.HttpsCertificateFilePath);
 
             if (this.UseHttps && string.IsNullOrWhiteSpace(this.HttpsCertificateFilePath))
                 throw new ConfigurationException("The path to a certificate needs to be provided when using https. Please use the argument 'certificatefilepath' to provide it.");
 
             var defaultApiHost = this.UseHttps
-                ? DefaultApiHost.Replace(@"http://", @"https://")
-                : DefaultApiHost;
+                ? defaults.ApiHost.Replace(@"http://", @"https://")
+                : defaults.ApiHost.Replace(@"https://", @"http://");
```

Comment:
Not a big deal but we could get UriBuilder to do this work for us.
```
var uriBuilder = new UriBuilder(defaultApiHost) {
	Scheme = this.UseHttps ? Uri.UriSchemeHttps : Uri.UriSchemeHttp
};
```

Reply:
Yeah I tried to keep changes to a minimum. Perhaps another PR can address this.